### ISSUE TITLE
Make fails 421bf3a608a4e0bcb1db391686c856e6bf52674d

### ISSUE THREAD

**Issue by v3ss0n** [rethinkdb/rethinkdb#3981]
when i  build 421bf3a608a4e0bcb1db391686c856e6bf52674d it fails with : 
```
[1/114] CC build/release/obj/rdb_protocol/protocol.o
src/rdb_protocol/protocol.cc: In member function ‘void rdb_r_unshard_visitor_t::unshard_range_batch(const query_t&, sorting_t)’:
src/rdb_protocol/protocol.cc:750:21: error: ‘base_exc_t’ was not declared in this scope
                     base_exc_t::GENERIC,
                     ^
src/rdb_protocol/protocol.cc:750:21: note: suggested alternative:
In file included from ./src/rdb_protocol/datum.hpp:23:0,
                 from ./src/perfmon/core.hpp:14,
                 from ./src/perfmon/perfmon.hpp:20,
                 from ./src/rdb_protocol/protocol.hpp:20,
                 from src/rdb_protocol/protocol.cc:2:
./src/rdb_protocol/error.hpp:19:7: note:   ‘ql::base_exc_t’
 class base_exc_t : public std::exception {
       ^
src/build.mk:396: recipe for target 'build/release/obj/rdb_protocol/protocol.o' failed
make[1]: *** [build/release/obj/rdb_protocol/protocol.o] Error 1
Makefile:52: recipe for target 'make' failed
make: *** [make] Error 2
```

**Comment by AtnNn** [comment]
177fb8a is a probable cause. Ping @mlucy.

**Comment by v3ss0n** [comment]
i tried building latest it fails more 565701f6649e04bfaf0dbe434e5a7150233d9ef1q
```
admin/build.mk:19: recipe for target 'build/web-assets' failed
make[1]: *** [build/web-assets] Error 8
Makefile:52: recipe for target 'make' failed
make: *** [make] Error 2
```
coz handlebars not found..
```
Error: Cannot find module 'handlebars'
  at Function.Module._resolveFilename (module.js:338:15)
  at Function.Module._load (module.js:280:25)
  at Module.require (module.js:364:17)
  at require (module.js:380:17)
```

i did ./configure --allow-fetch

**Comment by v3ss0n** [comment]
It seems that allowfetch is broken , i rebuild from clean but end up with more errors.
    [3/391] COFFEE build/drivers/javascript/coffee/cursor.js
/bin/bash: /home/v3ss/rethink_latest/build/external/coffee-script_1.7.1/bin/coffee: No such file or directory
drivers/javascript/build.mk:25: recipe for target 'build/drivers/javascript/coffee/cursor.js' failed

**Comment by AtnNn** [comment]
@v3ss0n Some changes were made to the build system today. Re-running `./configure` should solve the issues with coffee and handlebars.

**Comment by v3ss0n** [comment]
I tried that.
```
make clean
./configure --allow-fetch
make
```
same prob.

**Comment by AtnNn** [comment]
@v3ss0n I was able to get the same error. I'll try to fix it.

As a work-around, you can install coffee-script and browserify: `npm install -g coffee-script@1.7.1 browserify@3.24.13`

**Comment by v3ss0n** [comment]
Thats cool , its ok i am going to sleep for now , i was free for a bit and going to test tornado + rethinkdb async driver. 
I will try tomorrow when i get time.

**Comment by AtnNn** [comment]
The issue with fetching coffee-script should be fixed by 5793f17

**Comment by AtnNn** [comment]
As you can tell, the `next` branch is in active development.

Please consider using the 2.0 release candidate (based on the `v2.0.x` branch) to test the new async features of the Python driver:

http://download.rethinkdb.com/dev/2.0.0-0RC1/rethinkdb-2.0.0-0RC1.tgz

**Comment by v3ss0n** [comment]
Ok i wanted to test with latest RqlCursorEpmpty exception .i understand this branch is unstable.

**Comment by Tryneus** [comment]
@v3ss0n, you should be able to use the driver from the `v2.0.x` branch - Tornado integration wasn't included in the 2.0.0 RC1 release of the driver, but it should be compatible with that version of the server.

**Comment by v3ss0n** [comment]
Ok , i give up , fetching CoffeeScript is working with that branch but fail at handlebars again.
I will try with 2.0.x branch now.

**Comment by deontologician** [comment]
OK, the handlebars fix is in now. We are seeing next build successfully again. @v3ss0n if you pull next again to try to build it you'll probably want to make clean and reconfigure. There is some heavy caching because the npm modules take so long to redownload entirely.

**Comment by deontologician** [comment]
It looks like the compilation error only happens in release mode, not with `make DEBUG=1`, so that's a partial hint.

**Comment by VeXocide** [comment]
This is fixed in `next`, `v2.0.x`, and `v1.16.x` by commits  de7403e68146120459cd9bc7c3a3ab1cd6ec3dd2, 4594b2ca1187098602c801a5f6f6b37fda8002e8, and 8ee56e6435dacdd7353b685a1ea260ded9a8010a respectively.

Over the shoulder review by @danielmewes.
